### PR TITLE
Enable pylint's `dangerous-default-value` check

### PR DESCRIPTION
**Pull request by garrison**
When I ran pylint in its default configuration, I noticed a few `dangerous-default-value` warnings in the new cut finder.  I think it's worth adding this to our pylint configuration and fixing the existing issues in the code base.  (Even though users aren't supposed to directly call this code, adding this to our config allows us to prevent similar dangerous defaults in places we _do_ care about.)

### CODE REVIEW

**Review by ibrahim-shehzad — APPROVED** [review]
Interesting! I had to look at the examples given [here](https://docs.python.org/3/tutorial/controlflow.html#default-argument-values) to understand what the potential danger was but this makes sense now. This is super important. I made a minor modification to one of the tests and a docstring, just to make sure everything was consistent. This looks good to me overall, thanks

File: test/cutting/cut_finding/test_circuit_interfaces.py [hunk 1]
```diff
@@ -68,7 +68,7 @@ def test_circuit_conversion(self):
         assert max_wire_cuts_gamma(7) == 2
 
         # Assign by hand a different qubit mapping by specifiying init_qubit_names.
-        circuit_converted = SimpleGateList(trial_circuit, ["q0", "q1"])
+        circuit_converted = SimpleGateList(trial_circuit, ("q0", "q1"))
```
Comment (garrison) [inline test/cutting/cut_finding/test_circuit_interfaces.py:71]: I think this test could stay as it was, actually.  It's okay to pass a `list`; it's just not okay when the default function argument is a `list`.

Comment (ibrahim-shehzad) [reply test/cutting/cut_finding/test_circuit_interfaces.py:71]: Yes but I just figured this would be "more" consistent?

Comment (ibrahim-shehzad) [reply test/cutting/cut_finding/test_circuit_interfaces.py:71]: Your call though.

**Pull request merged**